client/lib/route/path.ts

```diff
@@ -45,7 +45,7 @@ export function getSiteFragment( path: URLString ): SiteSlug | SiteId | false {
 	for ( let i = 2; i > 0; i-- ) {
 		const piece = pieces[ pieces.length - i ];
 		if ( piece && -1 !== piece.indexOf( '.' ) ) {
-			return piece;
+			return piece.endsWith( '::' ) ? piece.replace( '::', '' ) : piece;
```

Comment:
Is there any chance that a given `piece` would have multiple occurrences of a double-colon? If so, it's important to note that `piece.replace` will only replace the first instance. To replace only a trailing instance, you'd want something like this:

```suggestion
			return piece.replace( /::$/, '' );
```

----

For context, try this in your browser's console:

```js
let test = 'abc::def::';
console.log( test.replace( '::', '' ) ); // 'abcdef::'
console.log( test.replace( /::$/, '' ) ); // 'abc::def'
```

Reply:
I can't think of a valid reason to have multiple double-colons, but I like the idea of keeping the fix specific to the bug, so I've implemented your suggestion. Thanks!